### PR TITLE
rpm: doc: Remove bundled SPEC and update README

### LINKED ISSUE #4257: Packaging, distro support and QA for existing RPM spec
I'd like to suggest and discuss a possible process to handle the distro availability and accompanying topics concerning the RPM spec. Last time, when I went to contribute support for several things in the existing RPM spec, it turned out the spec is not actually used for builds by the cloud-hypervisor org. There was also a suggestion to possibly exclude the spec into a separate repository.

Some experiment with help of [Open Build Service](https://build.opensuse.org/) has shown a possibility to have a separate repo on GitHub to carry the spec, while using OBS to actually build the package. OBS would also allow to reuse the same spec to support various RPM based distributions and create repositories for them. It can also build based on DEB specs, however this possibility might need to be explored more (perhaps launchpad might be another option for DEBs).

The components used for the experiment:
- OBS
  - Project and package definitions on OBS: https://build.opensuse.org/package/show/home:weltling:cloud-hypervisor/cloud-hypervisor
  - Note the `_service` file which among other tasks can carry fetch instructions
- Github 
 - https://github.com/weltling/cloud-hypervisor-obs.git

A repo generated for OpenSUSE:
https://download.opensuse.org/repositories/home:/weltling:/cloud-hypervisor/openSUSE_Tumbleweed/

Thus, it seems possible to use OBS and Github to produce packages either for QA or for production use. This can be done in a transparent manner, so then
- The latest Cloud Hypervisor releases can be made available through a repository
- Other pieces can be provided, too (fe EDK2, rust firmware, etc.)

Pro:
- Expanding the availability of the Cloud Hypervisor project
- Simple access to all needed pieces without having to compile everything by hand or dig in github repos
- Additional QA for RPMs and other specs
- Building and running on different distributions can reveal more bugs
- Specs being on github are easy to be contributed to

Contra:
- Additional effort maintaining github repo
- Additional effort maintaining build service

If this is found useful, I'd sure step up for the OBS and Github parts. OBS can have multiple users on projects, so that can be handled at the org level, too.

Thanks

### PR DESCRIPTION
**Pull request by weltling**
Packages targeting various distributions are now available thanks to the
Open Bulid Service. The SPECS have been moved to a separate repo under

https://github.com/cloud-hypervisor/obs-packaging

which documents how to include pre-compiled binaires in a supported
distribution. Besides the Cloud Hypervisor itself, accompanying pieces
like UEFI bins are available, too.

Any contributions to the packaging can be also accepted in
the aforementioned repo. Further integration with the OBS like PR
validation is to be targeted there, too, as far as it can get. As the
packaging topic has moved in a separate repo, the bundled SPEC became
obsolete and is suggested for removal.

Fixes: https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4257

Signed-off-by: Anatol Belski <anbelski@linux.microsoft.com>
Co-authored-by: Sebastien Boeuf <sebastien.boeuf@intel.com>

### CODE REVIEW

**Review by likebreath — APPROVED** [review]
Thank you for the contribution. LGTM. 

Only one nit for the commit message: please use `Fixes: #4257` if that issue is resolved by this PR.

**Comment by weltling** [comment]
ACK, changed as suggested in both commit message and PR description. Thanks for the checks!

File: README.md [hunk 1]
```diff
@@ -152,6 +153,15 @@ $ ./scripts/dev_cli.sh tests --unit
...
 development script commands and their related options.
 
+## Use Prebuilt Binaries
+
+Cloud Hypervisor packages targeting some popular Linux distributions are available
+thanks to the [Open Build Service](https://build.opensuse.org). The
+[OBS README](https://github.com/cloud-hypervisor/obs-packaging) explains how to
+enable the repository in a supported Linux distribution and install Cloud Hypervisor
+and accompanying packages. Please report any packaging issues in the
+[obs-packaging](https://github.com/cloud-hypervisor/obs-packagin) repository.
```
Comment (sboeuf) [inline README.md:163]:
```suggestion
[obs-packaging](https://github.com/cloud-hypervisor/obs-packaging) repository.
```

File: README.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
   - [Install prerequisites](#install-prerequisites)
   - [Clone and build](#clone-and-build)
     - [Containerized builds and tests](#containerized-builds-and-tests)
+  - [Use Prebuilt Binaries](#use-obs-packages)
```
Comment (sboeuf) [inline README.md:12]:
```suggestion
- [Use Prebuilt Binaries](#use-prebuilt-binaries)
```

**Review by sboeuf — CHANGES_REQUESTED** [review]
Could you please provide an extensive commit message explaining why we're removing the SPEC files from this repo. I know we can find the information in the Github issue, but from a `git log` perspective it's always good to be able to understand the rationale behind a patch.

**Comment by weltling** [comment]
Thanks for the review. I've applied the fixes, rebased and put more info in the commit message.

Thanks

**Pull request merged**